src/sentry/issues/group_policy.py

```diff
@@ -0,0 +1,167 @@
+                )
+            prev_limit = self.early_access.ignore_limit
+
+        if self.default.ignore_limit < prev_limit:
+            raise ValueError(
+                "Default ignore limit must be lower than Early Access and Limited Access organization limits"
+            )
+
+
+def get_noise_config(group_policy: Type[GroupPolicy], organization: Organization) -> NoiseConfig:
+    if group_policy.organizations and organization.slug in group_policy.organizations.keys():
+        return group_policy.organizations[organization.slug]
+
+    if organization.flags.early_adopter.is_set and group_policy.early_access:
+        return group_policy.early_access
+
+    return group_policy.default
+
+
+def get_group_policy_by_type_id(type_id: int) -> Type[GroupPolicy]:
+    if type_id not in _group_policy_registry:
+        raise ValueError(f"No group policy with the id {type_id} is registered.")
+    return _group_policy_registry[type_id]
+
+
+def reduce_noise(
+    new_grouphashes: Set[str],
+    performance_problems_by_hash: Dict[str, PerformanceProblem],
+    project: Project,
+) -> Set[str]:
+    groups_to_ignore = set()
+    cluster_key = settings.SENTRY_PERFORMANCE_ISSUES_RATE_LIMITER_OPTIONS.get("cluster", "default")
+    client = redis.redis_clusters.get(cluster_key)
+
+    for new_grouphash in new_grouphashes:
+        group_type = performance_problems_by_hash[new_grouphash].type
+        group_policy = get_group_policy_by_type_id(group_type.type_id)
+        # use default policy if none exists
+        if not group_policy:
+            noise_config = NoiseConfig()
+        else:
+            noise_config = get_noise_config(group_policy, project.organization)
+
+        if not should_create_group(client, new_grouphash, group_type, noise_config, project):
+            groups_to_ignore.add(new_grouphash)
+
+    new_grouphashes = new_grouphashes - groups_to_ignore
+    return new_grouphashes
+
+
+@metrics.wraps("performance.performance_issue.should_create_group", sample_rate=1.0)
+def should_create_group(
+    client: Any, grouphash: str, grouptype: GroupType, noise_config: NoiseConfig, project: Project
+) -> bool:
+    key = f"grouphash:{grouphash}:{project.id}"
+    times_seen = client.incr(key)
+
+    metrics.incr(
+        "performance.performance_issue.grouphash_counted",
+        tags={
+            "times_seen": times_seen,
+            "group_type": grouptype.slug,
+        },
+        sample_rate=1.0,
+    )
+
+    if times_seen >= noise_config.ignore_limit:
+        client.delete(grouphash)
+        metrics.incr(
+            "performance.performance_issue.issue_will_be_created",
+            tags={"group_type": grouptype.slug},
+            sample_rate=1.0,
+        )
+
+        return True
+    else:
+        client.expire(key, noise_config.expiry_time)  # 24 hour expiration from last seen
+        return False
+
+
+@dataclass(frozen=True)
+class ErrorPolicy(GroupPolicy):
+    group_type_id = 1
```

Comment:
Is there a way to do `ErrorGroupType.type_id` here (and for the other subclasses) so it's only hard coded in one place?